test/test_distributions.py

```diff
@@ -2944,6 +3064,41 @@ def test_kl_multivariate_normal_batched(self):
                                   MultivariateNormal(loc[1], scale_tril=scale_tril[1]))
         self.assertEqual(expected_kl, actual_kl)
 
+    def test_kl_lowrank_multivariate_normal(self):
+        set_rng_seed(0)  # see Note [Randomized statistical tests]
+        n = 5  # Number of tests for lowrank_multivariate_normal
+        for i in range(0, n):
+            loc = [torch.randn(4) for _ in range(0, 2)]
+            scale_factor = [torch.randn(4, 3) for _ in range(0, 2)]
+            scale_diag = [transform_to(constraints.positive)(torch.randn(4)) for _ in range(0, 2)]
+            covariance_matrix = [scale_factor[i].matmul(scale_factor[i].t()) +
+                                 scale_diag[i].diag() for i in range(0, 2)]
+            p = LowRankMultivariateNormal(loc[0], scale_factor[0], scale_diag[0])
+            q = LowRankMultivariateNormal(loc[1], scale_factor[1], scale_diag[1])
+            actual = kl_divergence(p, q)
```

Comment:
Is it also worth testing `kl_divergence(p_full, q)` and `kl_divergence(p, q_full)`?

Reply:
I didn't implement kl of lowrank_mvn vs mvn. I will try to write down the math tomorrow and implement it. Then we can test. :)